src/web/app/pages/ConnectMachineIp/index.tsx

```diff
@@ -153,56 +155,134 @@ const ConnectMachineIp = (): JSX.Element => {
   };
 
   const testCamera = async (device: IDeviceInfo) => {
-    setState((prev) => ({ ...prev, testState: TestState.CAMERA_TESTING }));
+    updateTestState({ testState: TestState.CAMERA_TESTING });
+
     const res = await checkCamera(device);
 
-    setState((prev) => ({
-      ...prev,
-      testState: res ? TestState.TEST_COMPLETED : TestState.CAMERA_TEST_FAILED,
-    }));
+    if (res) {
+      updateTestState({ testState: TestState.TEST_COMPLETED });
+
+      return;
+    }
+
+    updateTestState({ testState: TestState.CAMERA_TEST_FAILED });
+    alertCaller.popUp({
+      messageIcon: 'warning',
+      caption: lang.initialize.connect_machine_ip.check_camera,
+      message: isPromark ? lang.web_cam.no_device : lang.topbar.alerts.fail_to_connect_with_camera,
+    });
+  };
+
+  const checkSwiftrayConnection = () => {
+    updateTestState({ testState: TestState.IP_TESTING });
+
+    if (swiftrayReadyState === WebSocket.OPEN) {
+      return true;
+    }
+
+    if (swiftrayReadyState === WebSocket.CLOSED) {
+      updateTestState({ testState: TestState.IP_UNREACHABLE });
+      alertCaller.popUp({
+        messageIcon: 'warning',
+        caption: lang.initialize.connect_machine_ip.check_swiftray_connection,
+        message: lang.initialize.connect_machine_ip.alert.swiftray_connection_error,
+      });
+    }
+
+    return false;
+  };
+
+  const handleStartTestForPromark = async () => {
+    if (!checkSwiftrayConnection()) return;
+
+    const device = await testConnection(({ model }) => promarkModels.has(model));
+    if (!device) {
+      alertCaller.popUp({
+        messageIcon: 'warning',
+        caption: lang.initialize.connect_machine_ip.check_connection,
+        message: lang.message.unknown_device,
+      });
+      return;
+    }
+
+    await testCamera(device);
   };
 
   const handleStartTest = async () => {
+    if (isPromark) {
+      handleStartTestForPromark();
+      return;
+    }
+
     const { testState } = state;
     if (isTesting(testState)) return;
-    if (!testIpFormat()) return;
+    if (!validateIpFormat()) return;
+
     const ip = await testIpReachability();
     if (!ip) return;
+
     setUpLocalStorageIp(ip);
+
     testingIps.forEach((testingIp) => {
       discoverer.poke(testingIp);
       discoverer.pokeTcp(testingIp);
       discoverer.testTcp(testingIp);
     });
+
     const device = await testConnection();
     if (!device) return;
+
     if (!checkSoftwareForAdor(device)) {
       setIpValue('');
       setState((prev) => ({ ...prev, device: null, testState: TestState.NONE }));
       return;
     }
-    testCamera(device);
+
+    await testCamera(device);
   };
 
+  useEffect(() => {
+    checkRpiIp().then((ip) => ip && setIpValue(ip));
```

Comment:
Can add `if (!isUSb)` to this line since the RPi ip is only for network connection